microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/properties/PropertiesFileValidator.java

```diff
@@ -419,6 +420,16 @@ private static boolean isBuildtimePlaceholder(String str) {
 		return str.startsWith("${") && str.endsWith("}");
 	}
 
+	/**
+	 * Return true if the variable is an ENV variable, and false otherwise
+	 *
+	 * @param variable the variable to check
+	 * @return true if the variable is an ENV variable, and false otherwise
+	 */
+	private static boolean isEnvVariable(String variable) {
+		return variable.toUpperCase().equals(variable);

```

Comment:
variable.toUpperCase() will create a new instance of String (memory problem).

Why you could not do write like this:

```java
public static boolean isEnvVariable(String variable) {
	char[] chars = variable.toCharArray();
	for (char c : chars) {
		if (Character.isLetter(c)) {
			return Character.isUpperCase(c);
		}
	}
	return true;
}
```

I think it should be better to move this method in a new class like` com.redhat.qute.utils.EnvUtils` and write several test with this method. It i simportant to have those kind of test because if we change the implementation or the logic we will be sure that we don't break anything.

Reply:
This would be in lsp4mp correct?

Reply:
Ooops sorry, I wanted to say `org.eclipse.lsp4mp.utils.EnvUtils`